### ISSUE TITLE
firebase-crashlytics-ndk Native crash not reported in release build

### ISSUE THREAD

**Issue by guillaume-tgl** [firebase/firebase-android-sdk#1756]
### [REQUIRED] Step 2: Describe your environment

  * Android Studio version: 4.0
  * Firebase Component: Crashlytics + Crashlytics NDK
  * Component version: 17.1.0

### [REQUIRED] Step 3: Describe the problem

I'm testing if native and java crashes are correctly reported on debug and release builds of my app. All of them are correctly reported except for native crashes in a release build.

Here is the log file after a native crash in a debug build: 

```
2020-07-03 11:57:02.115 31711-31711/com.thegrizzlylabs.geniusscan.free.debug D/FirebaseCrashlytics: Firebase Analytics is available.
2020-07-03 11:57:02.116 31711-31711/com.thegrizzlylabs.geniusscan.free.debug D/FirebaseCrashlytics: Firebase Analytics listener registered successfully.
2020-07-03 11:57:02.141 31711-31711/com.thegrizzlylabs.geniusscan.free.debug D/FirebaseCrashlytics: Reading cached settings...
2020-07-03 11:57:02.147 31711-31711/com.thegrizzlylabs.geniusscan.free.debug D/FirebaseCrashlytics: Loaded cached settings: {"settings_version":3,"cache_duration":86400,"features":{"collect_logged_exceptions":true,"collect_reports":true,"collect_analytics":false,"prompt_enabled":false,"push_enabled":false,"firebase_crashlytics_enabled":false},"app":{"status":"activated","update_required":true,"report_upload_variant":1,"native_report_upload_variant":2},"fabric":{"org_id":"4f5e6d8af69f76766700162c","bundle_id":"com.thegrizzlylabs.geniusscan.free.debug"},"expires_at":1593770472315}
2020-07-03 11:57:02.147 31711-31711/com.thegrizzlylabs.geniusscan.free.debug D/FirebaseCrashlytics: Returning cached settings.
2020-07-03 11:57:02.148 31711-31711/com.thegrizzlylabs.geniusscan.free.debug D/FirebaseCrashlytics: Mapping file ID is: d029b2cc333d4576a934ccf52efc45dc
2020-07-03 11:57:02.148 31711-31711/com.thegrizzlylabs.geniusscan.free.debug I/FirebaseCrashlytics: Initializing Crashlytics 17.1.0
2020-07-03 11:57:02.149 31711-31711/com.thegrizzlylabs.geniusscan.free.debug D/FirebaseCrashlytics: Installer package name is: null
2020-07-03 11:57:02.172 31711-31711/com.thegrizzlylabs.geniusscan.free.debug D/FirebaseCrashlytics: Exception handling initialization successful
2020-07-03 11:57:02.179 31711-31823/com.thegrizzlylabs.geniusscan.free.debug D/FirebaseCrashlytics: Found matching FID, using Crashlytics IID: 6022849938c447d3a43af3c49de6f311
2020-07-03 11:57:02.179 31711-31823/com.thegrizzlylabs.geniusscan.free.debug D/FirebaseCrashlytics: Opening a new session with ID 5EFF00EE00AC-0001-7BDF-43E2ECEFFA7C
2020-07-03 11:57:02.183 31711-31823/com.thegrizzlylabs.geniusscan.free.debug D/libcrashlytics: Initializing libcrashlytics version 2.0.5
2020-07-03 11:57:02.183 31711-31823/com.thegrizzlylabs.geniusscan.free.debug D/libcrashlytics: Initializing native crash handling successful.
2020-07-03 11:57:02.183 31711-31823/com.thegrizzlylabs.geniusscan.free.debug I/FirebaseCrashlytics: Crashlytics NDK initialization successful
2020-07-03 11:57:02.205 31711-31823/com.thegrizzlylabs.geniusscan.free.debug D/FirebaseCrashlytics: Initialization marker file created.
2020-07-03 11:57:02.207 31711-31823/com.thegrizzlylabs.geniusscan.free.debug D/FirebaseCrashlytics: Registered Firebase Analytics event receiver for breadcrumbs
2020-07-03 11:57:02.207 31711-31823/com.thegrizzlylabs.geniusscan.free.debug D/FirebaseCrashlytics: Finalizing previously open sessions.
2020-07-03 11:57:02.215 31711-31823/com.thegrizzlylabs.geniusscan.free.debug D/FirebaseCrashlytics: Finalizing native report for session 5EFEFB64022B-0001-636E-43E2ECEFFA7C
2020-07-03 11:57:02.416 31711-31823/com.thegrizzlylabs.geniusscan.free.debug D/FirebaseCrashlytics: Could not generate ID for file /memfd:/jit-cache (deleted)
    java.io.FileNotFoundException: /memfd:/jit-cache (deleted): open failed: ENOENT (No such file or directory)
        at libcore.io.IoBridge.open(IoBridge.java:496)
        at java.io.FileInputStream.<init>(FileInputStream.java:159)
        at java.io.FileInputStream.<init>(FileInputStream.java:115)
        at com.google.firebase.crashlytics.c.k.e.b(Sha1FileIdStrategy.java:37)
        at com.google.firebase.crashlytics.c.k.e.a(Sha1FileIdStrategy.java:28)
        at com.google.firebase.crashlytics.c.k.a.g(BinaryImagesConverter.java:114)
        at com.google.firebase.crashlytics.c.k.a.h(BinaryImagesConverter.java:65)
        at com.google.firebase.crashlytics.c.k.a.a(BinaryImagesConverter.java:55)
        at com.google.firebase.crashlytics.c.k.b.a(NativeFileUtils.java:38)
        at com.google.firebase.crashlytics.c.h.k.b0(CrashlyticsController.java:1853)
        at com.google.firebase.crashlytics.c.h.k.S(CrashlyticsController.java:1122)
        at com.google.firebase.crashlytics.c.h.k.N(CrashlyticsController.java:858)
        at com.google.firebase.crashlytics.c.h.k.T(CrashlyticsController.java:793)
        at com.google.firebase.crashlytics.c.h.m.i(CrashlyticsCore.java:222)
        at com.google.firebase.crashlytics.c.h.m.a(CrashlyticsCore.java:43)
        at com.google.firebase.crashlytics.c.h.m$a.a(CrashlyticsCore.java:197)
        at com.google.firebase.crashlytics.c.h.m$a.call(CrashlyticsCore.java:194)
        at com.google.firebase.crashlytics.c.h.j0$c.run(Utils.java:134)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
        at com.google.firebase.crashlytics.c.h.v$a$a.a(ExecutorUtils.java:60)
        at com.google.firebase.crashlytics.c.h.d.run(BackgroundPriorityRunnable.java:27)
        at java.lang.Thread.run(Thread.java:919)
     Caused by: android.system.ErrnoException: open failed: ENOENT (No such file or directory)
        at libcore.io.Linux.open(Native Method)
        at libcore.io.ForwardingOs.open(ForwardingOs.java:167)
        at libcore.io.BlockGuardOs.open(BlockGuardOs.java:252)
        at libcore.io.ForwardingOs.open(ForwardingOs.java:167)
        at android.app.ActivityThread$AndroidOs.open(ActivityThread.java:8002)
        at libcore.io.IoBridge.open(IoBridge.java:482)
        at java.io.FileInputStream.<init>(FileInputStream.java:159) 
        at java.io.FileInputStream.<init>(FileInputStream.java:115) 
        at com.google.firebase.crashlytics.c.k.e.b(Sha1FileIdStrategy.java:37) 
        at com.google.firebase.crashlytics.c.k.e.a(Sha1FileIdStrategy.java:28) 
        at com.google.firebase.crashlytics.c.k.a.g(BinaryImagesConverter.java:114) 
        at com.google.firebase.crashlytics.c.k.a.h(BinaryImagesConverter.java:65) 
        at com.google.firebase.crashlytics.c.k.a.a(BinaryImagesConverter.java:55) 
        at com.google.firebase.crashlytics.c.k.b.a(NativeFileUtils.java:38) 
        at com.google.firebase.crashlytics.c.h.k.b0(CrashlyticsController.java:1853) 
        at com.google.firebase.crashlytics.c.h.k.S(CrashlyticsController.java:1122) 
        at com.google.firebase.crashlytics.c.h.k.N(CrashlyticsController.java:858) 
        at com.google.firebase.crashlytics.c.h.k.T(CrashlyticsController.java:793) 
        at com.google.firebase.crashlytics.c.h.m.i(CrashlyticsCore.java:222) 
        at com.google.firebase.crashlytics.c.h.m.a(CrashlyticsCore.java:43) 
        at com.google.firebase.crashlytics.c.h.m$a.a(CrashlyticsCore.java:197) 
        at com.google.firebase.crashlytics.c.h.m$a.call(CrashlyticsCore.java:194) 
        at com.google.firebase.crashlytics.c.h.j0$c.run(Utils.java:134) 
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167) 
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641) 
        at com.google.firebase.crashlytics.c.h.v$a$a.a(ExecutorUtils.java:60) 
        at com.google.firebase.crashlytics.c.h.d.run(BackgroundPriorityRunnable.java:27) 
        at java.lang.Thread.run(Thread.java:919) 
2020-07-03 11:57:02.849 31711-31824/com.thegrizzlylabs.geniusscan.free.debug D/FirebaseCrashlytics: Received Analytics message: 3 Bundle[{params=Bundle[{_o=auto, _sc=MainActivity, _si=6168665889145382641}], name=_vs, timestampInMillis=1593770222821}]
2020-07-03 11:57:02.849 31711-31825/com.thegrizzlylabs.geniusscan.free.debug D/FirebaseCrashlytics: Server says an update is required - forcing a full App update.
2020-07-03 11:57:02.856 31711-31825/com.thegrizzlylabs.geniusscan.free.debug D/FirebaseCrashlytics: Sending app info to https://update.crashlytics.com/spi/v1/platforms/android/apps/com.thegrizzlylabs.geniusscan.free.debug
2020-07-03 11:57:03.525 31711-31825/com.thegrizzlylabs.geniusscan.free.debug D/FirebaseCrashlytics: Update app request ID: fb2cfec501c6029ef8b2117274f161d1
2020-07-03 11:57:03.525 31711-31825/com.thegrizzlylabs.geniusscan.free.debug D/FirebaseCrashlytics: Result was 204
2020-07-03 11:57:03.792 31711-31823/com.thegrizzlylabs.geniusscan.free.debug D/FirebaseCrashlytics: Closing open sessions.
2020-07-03 11:57:03.792 31711-31823/com.thegrizzlylabs.geniusscan.free.debug D/FirebaseCrashlytics: Closing session: 5EFEFB64022B-0001-636E-43E2ECEFFA7C
2020-07-03 11:57:03.792 31711-31823/com.thegrizzlylabs.geniusscan.free.debug D/FirebaseCrashlytics: Collecting session parts for ID 5EFEFB64022B-0001-636E-43E2ECEFFA7C
2020-07-03 11:57:03.792 31711-31823/com.thegrizzlylabs.geniusscan.free.debug D/FirebaseCrashlytics: Session 5EFEFB64022B-0001-636E-43E2ECEFFA7C has fatal exception: false
2020-07-03 11:57:03.793 31711-31823/com.thegrizzlylabs.geniusscan.free.debug D/FirebaseCrashlytics: Session 5EFEFB64022B-0001-636E-43E2ECEFFA7C has non-fatal exceptions: false
2020-07-03 11:57:03.793 31711-31823/com.thegrizzlylabs.geniusscan.free.debug D/FirebaseCrashlytics: No events present for session ID 5EFEFB64022B-0001-636E-43E2ECEFFA7C
2020-07-03 11:57:03.793 31711-31823/com.thegrizzlylabs.geniusscan.free.debug D/FirebaseCrashlytics: Removing session part files for ID 5EFEFB64022B-0001-636E-43E2ECEFFA7C
2020-07-03 11:57:03.795 31711-31823/com.thegrizzlylabs.geniusscan.free.debug D/FirebaseCrashlytics: Closed all previously open sessions
2020-07-03 11:57:03.796 31711-31823/com.thegrizzlylabs.geniusscan.free.debug D/FirebaseCrashlytics: Unsent reports are available.
2020-07-03 11:57:03.797 31711-31823/com.thegrizzlylabs.geniusscan.free.debug D/FirebaseCrashlytics: Automatic data collection is enabled. Allowing upload.
2020-07-03 11:57:03.911 31711-31823/com.thegrizzlylabs.geniusscan.free.debug D/FirebaseCrashlytics: Initialization marker file removed: true
2020-07-03 11:57:03.911 31711-31823/com.thegrizzlylabs.geniusscan.free.debug D/FirebaseCrashlytics: Checking for crash reports...
2020-07-03 11:57:03.913 31711-31823/com.thegrizzlylabs.geniusscan.free.debug D/FirebaseCrashlytics: Reports are being sent.
2020-07-03 11:57:03.916 31711-31917/com.thegrizzlylabs.geniusscan.free.debug D/FirebaseCrashlytics: Logging Crashlytics event to Firebase
2020-07-03 11:57:03.916 31711-31917/com.thegrizzlylabs.geniusscan.free.debug D/FirebaseCrashlytics: Awaiting app exception callback from FA...
2020-07-03 11:57:03.918 31711-31823/com.thegrizzlylabs.geniusscan.free.debug D/FirebaseCrashlytics: Send via DataTransport disabled. Removing DataTransport reports.
2020-07-03 11:57:03.919 31711-31918/com.thegrizzlylabs.geniusscan.free.debug D/FirebaseCrashlytics: Starting report processing in 1.0 second(s)...
2020-07-03 11:57:03.960 31711-31824/com.thegrizzlylabs.geniusscan.free.debug D/FirebaseCrashlytics: Received Analytics message: 3 Bundle[{params=Bundle[{_o=clx, _r=1, _et=1119, _sc=MainActivity, _si=6168665889145382641, timestamp=1593770220000, fatal=1}], name=_ae, timestampInMillis=1593770223916}]
2020-07-03 11:57:03.960 31711-31917/com.thegrizzlylabs.geniusscan.free.debug D/FirebaseCrashlytics: App exception callback received from FA listener.
2020-07-03 11:57:04.014 31711-31824/com.thegrizzlylabs.geniusscan.free.debug D/FirebaseCrashlytics: Received Analytics message: 3 Bundle[{params=Bundle[{_o=auto, _et=1149, _fr=1, _sc=MainActivity, _si=6168665889145382641}], name=_e, timestampInMillis=1593770223967}]
2020-07-03 11:57:04.920 31711-31918/com.thegrizzlylabs.geniusscan.free.debug D/FirebaseCrashlytics: Attempting to send 1 report(s)
2020-07-03 11:57:04.924 31711-31918/com.thegrizzlylabs.geniusscan.free.debug D/FirebaseCrashlytics: Sending report to: https://reports.crashlytics.com/sdk-api/v1/platforms/android/apps/com.thegrizzlylabs.geniusscan.free.debug/minidumps
2020-07-03 11:57:05.804 31711-31918/com.thegrizzlylabs.geniusscan.free.debug D/FirebaseCrashlytics: Result was: 202
2020-07-03 11:57:05.805 31711-31918/com.thegrizzlylabs.geniusscan.free.debug I/FirebaseCrashlytics: Crashlytics Reports Endpoint upload complete: 5EFEFB64022B-0001-636E-43E2ECEFFA7C
2020-07-03 11:57:05.805 31711-31918/com.thegrizzlylabs.geniusscan.free.debug D/FirebaseCrashlytics: Removing native report file at /data/user/0/com.thegrizzlylabs.geniusscan.free.debug/files/.com.google.firebase.crashlytics/native-sessions/5EFEFB64022B-0001-636E-43E2ECEFFA7C/metadata
2020-07-03 11:57:05.805 31711-31918/com.thegrizzlylabs.geniusscan.free.debug D/FirebaseCrashlytics: Removing native report file at /data/user/0/com.thegrizzlylabs.geniusscan.free.debug/files/.com.google.firebase.crashlytics/native-sessions/5EFEFB64022B-0001-636E-43E2ECEFFA7C/user
2020-07-03 11:57:05.806 31711-31918/com.thegrizzlylabs.geniusscan.free.debug D/FirebaseCrashlytics: Removing native report file at /data/user/0/com.thegrizzlylabs.geniusscan.free.debug/files/.com.google.firebase.crashlytics/native-sessions/5EFEFB64022B-0001-636E-43E2ECEFFA7C/device
2020-07-03 11:57:05.806 31711-31918/com.thegrizzlylabs.geniusscan.free.debug D/FirebaseCrashlytics: Removing native report file at /data/user/0/com.thegrizzlylabs.geniusscan.free.debug/files/.com.google.firebase.crashlytics/native-sessions/5EFEFB64022B-0001-636E-43E2ECEFFA7C/minidump
2020-07-03 11:57:05.806 31711-31918/com.thegrizzlylabs.geniusscan.free.debug D/FirebaseCrashlytics: Removing native report file at /data/user/0/com.thegrizzlylabs.geniusscan.free.debug/files/.com.google.firebase.crashlytics/native-sessions/5EFEFB64022B-0001-636E-43E2ECEFFA7C/binaryImages
2020-07-03 11:57:05.806 31711-31918/com.thegrizzlylabs.geniusscan.free.debug D/FirebaseCrashlytics: Removing native report file at /data/user/0/com.thegrizzlylabs.geniusscan.free.debug/files/.com.google.firebase.crashlytics/native-sessions/5EFEFB64022B-0001-636E-43E2ECEFFA7C/logs
2020-07-03 11:57:05.807 31711-31918/com.thegrizzlylabs.geniusscan.free.debug D/FirebaseCrashlytics: Removing native report file at /data/user/0/com.thegrizzlylabs.geniusscan.free.debug/files/.com.google.firebase.crashlytics/native-sessions/5EFEFB64022B-0001-636E-43E2ECEFFA7C/session
2020-07-03 11:57:05.807 31711-31918/com.thegrizzlylabs.geniusscan.free.debug D/FirebaseCrashlytics: Removing native report file at /data/user/0/com.thegrizzlylabs.geniusscan.free.debug/files/.com.google.firebase.crashlytics/native-sessions/5EFEFB64022B-0001-636E-43E2ECEFFA7C/app
2020-07-03 11:57:05.807 31711-31918/com.thegrizzlylabs.geniusscan.free.debug D/FirebaseCrashlytics: Removing native report file at /data/user/0/com.thegrizzlylabs.geniusscan.free.debug/files/.com.google.firebase.crashlytics/native-sessions/5EFEFB64022B-0001-636E-43E2ECEFFA7C/os
2020-07-03 11:57:05.807 31711-31918/com.thegrizzlylabs.geniusscan.free.debug D/FirebaseCrashlytics: Removing native report directory at /data/user/0/com.thegrizzlylabs.geniusscan.free.debug/files/.com.google.firebase.crashlytics/native-sessions/5EFEFB64022B-0001-636E-43E2ECEFFA7C
```

And here is the log file after the same native crash in a release build:

```
2020-07-03 13:27:34.988 16081-16081/? D/FirebaseCrashlytics: Firebase Analytics is available.
2020-07-03 13:27:34.988 16081-16081/? D/FirebaseCrashlytics: Firebase Analytics listener registered successfully.
2020-07-03 13:27:34.995 16081-16081/? D/FirebaseCrashlytics: Reading cached settings...
2020-07-03 13:27:34.996 16081-16081/? D/FirebaseCrashlytics: Loaded cached settings: {"settings_version":3,"cache_duration":86400,"features":{"collect_logged_exceptions":true,"collect_reports":true,"collect_analytics":false,"prompt_enabled":false,"push_enabled":false,"firebase_crashlytics_enabled":false},"app":{"status":"activated","update_required":true,"report_upload_variant":1,"native_report_upload_variant":1},"fabric":{"org_id":"4f5e6d8af69f76766700162c","bundle_id":"com.thegrizzlylabs.geniusscan.free"},"expires_at":1593853462653}
2020-07-03 13:27:34.997 16081-16081/? D/FirebaseCrashlytics: Returning cached settings.
2020-07-03 13:27:34.997 16081-16081/? D/FirebaseCrashlytics: Mapping file ID is: 00000000000000000000000000000000
2020-07-03 13:27:34.998 16081-16081/? I/FirebaseCrashlytics: Initializing Crashlytics 17.1.0
2020-07-03 13:27:34.998 16081-16081/? D/FirebaseCrashlytics: Installer package name is: null
2020-07-03 13:27:35.005 16081-16081/? D/FirebaseCrashlytics: Exception handling initialization successful
2020-07-03 13:27:35.007 16081-16121/? D/FirebaseCrashlytics: Found matching FID, using Crashlytics IID: fb0da81551fe46e98d6bf0c955b40c67
2020-07-03 13:27:35.007 16081-16121/? D/FirebaseCrashlytics: Opening a new session with ID 5EFF16270005-0001-3ED1-0D26F2CF123D
2020-07-03 13:27:35.009 16081-16121/? D/libcrashlytics: Initializing libcrashlytics version 2.0.5
2020-07-03 13:27:35.010 16081-16121/? D/libcrashlytics: Initializing native crash handling successful.
2020-07-03 13:27:35.010 16081-16121/? I/FirebaseCrashlytics: Crashlytics NDK initialization successful
2020-07-03 13:27:35.020 16081-16121/? D/FirebaseCrashlytics: Initialization marker file created.
2020-07-03 13:27:35.020 16081-16121/? D/FirebaseCrashlytics: Registered Firebase Analytics event receiver for breadcrumbs
2020-07-03 13:27:35.020 16081-16121/? D/FirebaseCrashlytics: Finalizing previously open sessions.
2020-07-03 13:27:35.028 16081-16121/? D/FirebaseCrashlytics: Closing open sessions.
2020-07-03 13:27:35.028 16081-16121/? D/FirebaseCrashlytics: Closing session: 5EFF15FE0223-0001-3CEB-0D26F2CF123D
2020-07-03 13:27:35.028 16081-16121/? D/FirebaseCrashlytics: Collecting session parts for ID 5EFF15FE0223-0001-3CEB-0D26F2CF123D
2020-07-03 13:27:35.028 16081-16121/? D/FirebaseCrashlytics: Session 5EFF15FE0223-0001-3CEB-0D26F2CF123D has fatal exception: false
2020-07-03 13:27:35.028 16081-16121/? D/FirebaseCrashlytics: Session 5EFF15FE0223-0001-3CEB-0D26F2CF123D has non-fatal exceptions: false
2020-07-03 13:27:35.028 16081-16121/? D/FirebaseCrashlytics: No events present for session ID 5EFF15FE0223-0001-3CEB-0D26F2CF123D
2020-07-03 13:27:35.028 16081-16121/? D/FirebaseCrashlytics: Removing session part files for ID 5EFF15FE0223-0001-3CEB-0D26F2CF123D
2020-07-03 13:27:35.030 16081-16121/? D/FirebaseCrashlytics: Closed all previously open sessions
2020-07-03 13:27:35.031 16081-16121/? D/FirebaseCrashlytics: No reports are available.
2020-07-03 13:27:35.031 16081-16121/? D/FirebaseCrashlytics: Initialization marker file removed: true
2020-07-03 13:27:35.224 16081-16127/? D/FirebaseCrashlytics: Received Analytics message: 3 Bundle[{params=Bundle[{_o=auto, _sc=MainActivity, _si=4494261994654322968}], name=_vs, timestampInMillis=1593775655206}]
2020-07-03 13:27:35.228 16081-16125/? D/FirebaseCrashlytics: Server says an update is required - forcing a full App update.
2020-07-03 13:27:35.231 16081-16125/? D/FirebaseCrashlytics: Sending app info to https://update.crashlytics.com/spi/v1/platforms/android/apps/com.thegrizzlylabs.geniusscan.free
2020-07-03 13:27:35.803 16081-16125/? D/FirebaseCrashlytics: Update app request ID: 3c49ec7f075b39017e27fdd64a366286
2020-07-03 13:27:35.803 16081-16125/? D/FirebaseCrashlytics: Result was 204
```

**Comment by kevinkokomani** [comment]
Are there any difference between your debug and release builds that you know of? Have you ensured that your testing between debug and release is consistent in all other ways (Android version, device being tested)?

We've also seen this happen recently (native crashes not reporting for a particular configuration) and it was attributed to the way the actual crash itself was being triggered. Could you share the code you're using to crash the app? I might recommend using something like this to crash your app to test both scenarios above:

```
int* foo = (int*)-1;
    printf("%d\n", *foo);
    return *foo;
```

**Comment by guillaume-tgl** [comment]
Here is the error I get in Logcat in both cases:

```
2020-07-08 10:30:02.731 31156-31156/? A/DEBUG: *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
2020-07-08 10:30:02.731 31156-31156/? A/DEBUG: Build fingerprint: 'samsung/starltexx/starlte:10/QP1A.190711.020/G960FXXU9ETF5:user/release-keys'
2020-07-08 10:30:02.731 31156-31156/? A/DEBUG: Revision: '26'
2020-07-08 10:30:02.731 31156-31156/? A/DEBUG: ABI: 'arm64'
2020-07-08 10:30:02.736 31156-31156/? A/DEBUG: Timestamp: 2020-07-08 10:30:02+0200
2020-07-08 10:30:02.736 31156-31156/? A/DEBUG: pid: 15437, tid: 15437, name: geniusscan.free  >>> com.thegrizzlylabs.geniusscan.free <<<
2020-07-08 10:30:02.736 31156-31156/? A/DEBUG: uid: 10415
2020-07-08 10:30:02.736 31156-31156/? A/DEBUG: signal 6 (SIGABRT), code -6 (SI_TKILL), fault addr --------
2020-07-08 10:30:02.736 31156-31156/? A/DEBUG: Abort message: 'JNI DETECTED ERROR IN APPLICATION: obj == null
        in call to GetObjectField
        from com.thegrizzlylabs.geniusscan.sdk.ocr.OCREngine com.thegrizzlylabs.geniusscan.sdk.ocr.OCREngine$CppProxy.create(com.thegrizzlylabs.geniusscan.sdk.ocr.OCREngineConfiguration, com.thegrizzlylabs.geniusscan.sdk.core.Logger, com.thegrizzlylabs.geniusscan.sdk.ocr.OCREngineProgressListener)'
2020-07-08 10:30:02.736 31156-31156/? A/DEBUG:     x0  0000000000000000  x1  0000000000003c4d  x2  0000000000000006  x3  0000007febee5a70
2020-07-08 10:30:02.736 31156-31156/? A/DEBUG:     x4  fefeff7d9fdd1f97  x5  fefeff7d9fdd1f97  x6  fefeff7d9fdd1f97  x7  7f7f7f7fffff7fff
2020-07-08 10:30:02.736 31156-31156/? A/DEBUG:     x8  00000000000000f0  x9  0000007e9f8b75e0  x10 0000000000000000  x11 0000000000000001
2020-07-08 10:30:02.736 31156-31156/? A/DEBUG:     x12 000000001409af10  x13 ffffffffffffffff  x14 0000000000000000  x15 ffffffffffffffff
2020-07-08 10:30:02.736 31156-31156/? A/DEBUG:     x16 0000007e9f9858c0  x17 0000007e9f962880  x18 0000007ea5402000  x19 00000000000000ac
2020-07-08 10:30:02.736 31156-31156/? A/DEBUG:     x20 0000000000003c4d  x21 00000000000000b2  x22 0000000000003c4d  x23 00000000ffffffff
2020-07-08 10:30:02.736 31156-31156/? A/DEBUG:     x24 0000007ea44e0020  x25 0000000000000012  x26 0000000000000005  x27 0000007e1deb6cf5
2020-07-08 10:30:02.736 31156-31156/? A/DEBUG:     x28 0000007e1ded89d7  x29 0000007febee5b20
2020-07-08 10:30:02.736 31156-31156/? A/DEBUG:     sp  0000007febee5a50  lr  0000007e9f916330  pc  0000007e9f916360
```

The only differences between the debug build and release builds:

```
buildTypes {

        debug {
            multiDexEnabled true
            applicationIdSuffix ".debug"
            versionNameSuffix '-DEBUG'
        }

        release {
            minifyEnabled true
            proguardFiles getDefaultProguardFile('proguard-android.txt'), '../proguard-project.txt'
            signingConfig signingConfigs.release

            firebaseCrashlytics {
                nativeSymbolUploadEnabled true
                strippedNativeLibsDir "build/intermediates/stripped_native_libs/freeRelease/out/lib"
                unstrippedNativeLibsDir "build/intermediates/merged_native_libs/freeRelease/out/lib"
            }
        }
    }
```

I'll try to crash the app with your code to see if it changes something.

**Comment by kevinkokomani** [comment]
Thanks -- so we are able to repro this and we'll look into why this might be happening. In the meantime, I'd still want to see what happens with the other crash that I mentioned above, so looking forward to seeing your results on that.

Could you also try testing the original way that you're crashing the app, but trying it on some different Android versions (possibly Android 8 or 9)? We have a hunch it may report successfully in that case irrespective of the way you're crashing the app.

**Comment by guillaume-tgl** [comment]
I've tried with your code and this time, I see in the logs that the SDK is correctly uploading a report both in debug and release (`Crashlytics Reports Endpoint upload complete: 5F05E1E80372-0001-49E2-43E2ECEFFA7C`) but I can't see it in the Firebase console. When I throw a java exception in the same builds, it gets correctly reported. This is frustrating :|

This is on a Galaxy S9 with Android 10. I'll try on another device and let you know.

**Comment by guillaume-tgl** [comment]
Native crash on release build with my original way to crash is correctly reported on a Nexus 5X with Android 8 :)

**Comment by mrwillis21** [comment]
Related: #1678 - We're looking into this issue and will let you know when we have more info for you. :)

**Comment by mrwillis21** [comment]
@guillaume-tgl This is addressed in our latest release, Crashlytics NDK 17.3.0 (https://github.com/firebase/firebase-android-sdk/issues/1678#issuecomment-726349948). Please give that a shot and let us know how it goes! :)
